pkg/apis/vpc/v1alpha1/subnetconnectionbindingmap_types.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:validation:XValidation:rule="!(has(self.targetSubnetSet) && has(self.targetSubnet))",message="Only one of targetSubnetSet or targetSubnet can be specified"
+type SubnetConnectionBindingMapSpec struct {
+	// SubnetName is Subnet name which this binding is attached on.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="subnetName is immutable"
+	SubnetName string `json:"subnetName"`
+	// TargetSubnetSet specifies the target subnetSet which a subnet is bound to.
+	// +kubebuilder:validation:Optional
+	TargetSubnetSet string `json:"targetSubnetSet,omitempty"`
+	// TargetSubnet specifies the target subnet which a subnet is bound to.
+	// +kubebuilder:validation:Optional
+	TargetSubnet string `json:"targetSubnet,omitempty"`
+	// VLANTrafficTag is the vlan tag configured in the binding. It can be empty, then the handler will choose a valid
+	// value based on the existing subnet connection binding maps bound on the target subnet or subnetSet.
+	// +kubebuilder:validation:Maximum:=4095
+	// +kubebuilder:validation:Minimum:=1
+	VLANTrafficTag int64 `json:"vlanTrafficTag,omitempty"`
+}
+
+// SubnetConnectionBindingMapStatus defines the observed state of SubnetConnectionBindingMap.
+type SubnetConnectionBindingMapStatus struct {
+	Subnet      string                  `json:"subnet,omitempty"`
+	BindingMaps []ConnectionBindingMaps `json:"bindingMaps,omitempty"`
+	Conditions  []Condition             `json:"conditions,omitempty"`
+}
+
+type ConnectionBindingMaps struct {
```

Comment:
If we do not need this if we do not auto-allocate VLAN?

Reply:
I feel it is not necessary.

Reply:
Sure, I can remove it from the status.

Reply:
BTW, is it OK for you to show the CHILD VpcSubnet path in the status (https://github.com/vmware-tanzu/nsx-operator/pull/845/files#diff-bb38a914041e9190d7a2ba1e76acbecacac053613997e3957f682cf1c0c1d9b3R27) ? Or we shall show the CHILD Subnet CR name in the status?

Reply:
I see, removed.